app/notification.go

```diff
@@ -349,76 +349,156 @@ func sendNotificationEmail(post *model.Post, user *model.User, channel *model.Ch
 		// fall back to sending a single email if we can't batch it for some reason
 	}
 
-	var channelName string
-	var bodyText string
+	translateFunc := utils.GetUserTranslations(user.Locale)
```

Comment:
When I started, `sendNotificationEmail(...)` was one massive function that handled batching notifications, building non-batched emails, and sending those emails. I tried to split it out into some more reasonable parts to make it easier to reason about, and to make it easier to unit test. As mentioned in the PR description, I'm still working on unit tests for this file. It also has a lot of static function accesses that make this difficult.